Chore: (Docs) Writing stories section (#16236)
storybookjs/storybook: docs/writing-stories/args.md

title: 'Args'
---

> A story is a component with a set of arguments that define how the component should render. “Args” are Storybook’s mechanism for defining those arguments in a single JavaScript object. Args can be used to dynamically change props, slots, styles, inputs, etc. It allows Storybook and its addons to live edit components. You _do not_ need to modify your underlying component code to use args.

Reviewer: ```suggestion
A story is a component with a set of arguments that define how the component should render. “Args” are Storybook’s mechanism for defining those arguments in a single JavaScript object. Args can be used to dynamically change props, slots, styles, inputs, etc. This allows Storybook and its addons to live edit components. You _do not_ need to modify your underlying component code to use args.
```

"Args" is a strange term, grammatically. It's plural in its spelling, but singular in meaning. E.g. In "'Args' are Storybook's mechanism...", I think "is" would be technically correct, but it reads much more awkwardly than "are". Therefore, to be consistent with that decision, rather than "It allows Storybook...", we should use "This allows Storybook..." or "They allow Storybook...".

## Component args

> You can also define args at the component level; they will apply to all the component's stories unless you overwrite them. To do so, use the `arg` key on the `default` CSF export:

Reviewer: ```suggestion
You can also define args at the component level; they will apply to all the component's stories unless you overwrite them. To do so, use the `args` key on the `default` CSF export:
```

## Args composition

> You can separate the arguments to a story to compose in other stories.

Reviewer: I don't quite understand the use of "separate" here.

<div class="aside">

> 💡<strong>Note:</strong> If you find yourself applying this pattern often, you should consider using [component-level args](#component-args).

Reviewer: As this is MDX, we could have an Aside component that accepts a `title` prop to handle this formatting automatically. Not a blocker for this effort, but something to keep in mind if we find ourselves doing this often.

Reviewer: ```suggestion
💡<strong>Note:</strong> If you find yourself re-using the same args for most of a component's stories, you should consider using [component-level args](#component-args).
```

Or something like that. Basically, "pattern" is pretty broad term. One could "apply this pattern often" in a way that doesn't fit the recommendation. This change attempts to make the language more specific to avoid that.

## Args can modify any aspect of your component

> You can use args in your stories to configure the component's appearance, similar to what you would do in an application. For example, here's how you could use a `footer` arg to populate a child component:

Reviewer: I wonder if I more useful framing here is something like "you have full control over how `args` are applied to your component(s) in your story".

## Setting args through the URL

You can also override the set of initial args for the active story by adding an `args` query parameter to the URL. Typically you would use the Controls addon to handle this. For example, here's how you could set a `size` and `style` arg in the Storybook's URL:

```
?path=/story/avatar--default&args=style:rounded;size:100
```

> As a safety guard against [XSS](https://owasp.org/www-community/attacks/xss/) attacks, the arg's keys and values provided in the URL are limited to alphanumeric characters, spaces, underscores, and dashes. Any other types will be ignored and removed from the URL, but you can still use them with the Controls addon and code.

Reviewer: ```suggestion
As a safeguard against [XSS](https://owasp.org/www-community/attacks/xss/) attacks, the arg's keys and values provided in the URL are limited to alphanumeric characters, spaces, underscores, and dashes. Any other types will be ignored and removed from the URL, but you can still use them with the Controls addon and within your story.
```
